test/helper/zone-helper.spec.ts

```diff
@@ -0,0 +1,100 @@
+    const simpleStack = getNonZoneAwareStack(err);
+    if (!simpleStack) {
+      return;
+    }
+    const frames = simpleStack.split('\n');
+    for (let i = 0; i < frames.length; i++) {
+      expect(zoneAwareFrames.filter(f => frames[i].indexOf(f) !== -1).length).toBe(0);
+    }
+  };
+
+  const errorZoneSpec = {
+    name: 'errorZone',
+    done: null,
+    onHandleError: (parentDelegate, currentZone, targetZone, error) => {
+      expectStack(error);
+      setTimeout(() => {
+        errorZoneSpec.done && errorZoneSpec.done();
+      }, 0);
+      return false;
+    }
+  };
+
+  const errorZone = Zone.root.fork(errorZoneSpec);
+
+  const errorTest = function(testFn: Function) {
+    return function(done) {
+      errorZoneSpec.done = done;
+      errorZone.run(testFn);
+    };
+  };
+
+  it('test getNonZoneAwareStack', errorTest(() => {
+       setTimeout(() => {
+         throw new Error('test error');
+       }, 10);
+     }));
+
+  it('test promise getNonZoneAwareStack', (done) => {
+    const p = new Promise((resolve, reject) => {
+      reject(new Error('test error'));
+    });
+    p.catch(err => {
+      expectStack(err);
+      done();
+    });
+  });
+
+  it('test eventTask getNonZoneAwareStack', errorTest(() => {
+       const task = Zone.current.scheduleEventTask('errorEvent', () => {
+         throw new Error('test error');
+       }, null, () => null, null);
+       task.invoke();
+     }));
+
+  it('test longStackTrace getNonZoneAwareStack', errorTest(() => {
+       const task =
+           Zone.current.fork(Zone['longStackTraceZoneSpec']).scheduleEventTask('errorEvent', () => {
+             throw new Error('test error');
+           }, null, () => null, null);
+       task.invoke();
+     }));
+
+  it('test custom zoneSpec getNonZoneAwareStack', errorTest(() => {
+       // custom zone spec's stack trace should be keeped
+       const task = Zone.current.fork(Zone['longStackTraceZoneSpec'])
+                        .fork({
+                          name: 'customZone',
+                          onScheduleTask: (parentDelegate, currentZone, targetZone, task) => {
+                            return parentDelegate.scheduleTask(targetZone, task);
+                          },
+                          onHandleError: (parentDelegate, currentZone, targetZone, error) => {
+                            parentDelegate.handleError(targetZone, error);
+                            const containsCustomZoneSpecStackTrace =
+                                getNonZoneAwareStack(error).indexOf('onScheduleTask') !== -1;
+                            expect(containsCustomZoneSpecStackTrace).toBeTruthy();
+                            return false;
+                          }
+                        })
+                        .scheduleEventTask('errorEvent', () => {
+                          throw new Error('test error');
+                        }, null, () => null, null);
+       task.invoke();
```

Comment:
I write a help method errorTest which check the error stack, I will change the name of errorTest to assertStackDoesNotContainZoneFramesTest